src/babel/transformation/file/options.json

```diff
@@ -21,8 +21,8 @@
   },
 
   "moduleId": {
-    "description": "specify a custom name for module ids",
-    "type": "string"
+    "description": "specify a custom name for module ids, or a function to transform the default name",
+    "type": ["string", "function"]
```

Comment:
I don't really like overloading since it adds unnecessary complications and makes it awkward to document.